Add Waterfall Docs

finlaysawyer opened the pull request:
I'd expect you'll have a few changes to make, but hopefully this should suffice for initial Waterfall documentation. I figured it would be easier to navigate and more structured to rename /server to /paper-server and then adding /waterfall-server. Of course, I can change this if not desirable. I've updated all of the references for it.

Closes #32

electronicboy commented on source/index.rst:
```diff
@@ -39,7 +39,19 @@ configuring it, maintaining plugins, and general best practices.
-    server/index
+    paper-server/index
+
+Running A Waterfall Server
```
Waterfall is a proxy, not a server

electronicboy commented on source/waterfall-server/getting-started.rst:
```diff
@@ -0,0 +1,47 @@
+The amount of RAM can be set by changing the numbers in the ``-Xms`` and ``-Xmx``
+arguments.
+
+For more advanced Java tuning, see `Aikar's tuning page <https://mcflags.emc.gs>`_.
```
These flags aren't recommended for the proxy

finlaysawyer commented:
Thanks for the review, I've gone ahead and Implemented each of these - I added Aikar's flags separately underneath as this is essentially what's done with Paper as well.

Proximyst commented on source/waterfall/configuration.rst:
```diff
@@ -0,0 +1,51 @@
+disable_entity_metadata_rewrite
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+* **default**: false
+* **description**: Sets whether to disable entity metadata rewriting.
```
Perhaps we should explain what it does?

Chew replied:
What *does* it do?

Proximyst commented on source/waterfall/getting-started.rst:
```diff
@@ -0,0 +1,49 @@
+Requirements
+~~~~~~~~~~~~
+
+Waterfall and Travertine require **Java 8** or newer to run. It should run fine
```
```suggestion
Waterfall and Travertine require **Java 8** or newer to run. It does run fine
```

Proximyst commented on source/waterfall/getting-started.rst:
```diff
@@ -0,0 +1,49 @@
+
+Waterfall and Travertine require **Java 8** or newer to run. It should run fine
+on newer versions as well, such as Java 11, but certain plugins may have issues.
+Generally, LTS versions of Java are best supported, though you may have luck on newer versions.
```
```suggestion
Generally, LTS versions of Java are targeted, though you may have luck on newer versions.
```

Proximyst commented on source/waterfall/getting-started.rst:
```diff
@@ -0,0 +1,49 @@
+``java -Xms512M -Xmx512M -jar waterfall-###.jar``
+
+Aikar's recommended flags for Waterfall and Travertine are as follows:
+``java -Xms512M -Xmx512M -XX:+UseG1GC -XX:G1HeapRegionSize=4M -XX:+UnlockExperimentalVMOptions``
```
I'd say we should rather link to Aikar's article in case they change. It also has info on Pterodactyl/Docker's memory limits and whatnot.

Chew replied:
The flags aren't on the article (that I see), he put it in a pin in #waterfall-help I believe.

Proximyst replied:
Bully Aikar into adding it then :>

finlaysawyer replied:
Yeah I did have the article linked originally but electronicboy said to remove them as they're just for Paper, we can either keep them until/if they're added or just remove them for the time being?

Proximyst commented on source/waterfall/getting-started.rst:
```diff
@@ -0,0 +1,49 @@
+``java -Xms512M -Xmx512M -XX:+UseG1GC -XX:G1HeapRegionSize=4M -XX:+UnlockExperimentalVMOptions``
+``-XX:+ParallelRefProcEnabled -XX:+AlwaysPreTouch -jar waterfall-###.jar``
+
+The amount of RAM can be set by changing the numbers in the ``-Xms`` and ``-Xmx``
```
```suggestion
The amount of memory can be set by changing the numbers in the ``-Xms`` and ``-Xmx``
```

This is strictly speaking the heap memory for the JVM.

Proximyst commented on source/waterfall/getting-started.rst:
```diff
@@ -0,0 +1,49 @@
+``-XX:+ParallelRefProcEnabled -XX:+AlwaysPreTouch -jar waterfall-###.jar``
+
+The amount of RAM can be set by changing the numbers in the ``-Xms`` and ``-Xmx``
+arguments.
```
```suggestion
flags.
```

Proximyst commented on source/waterfall/getting-started.rst:
```diff
@@ -0,0 +1,49 @@
+The amount of RAM can be set by changing the numbers in the ``-Xms`` and ``-Xmx``
+arguments.
+
+To configure your server, see the :doc:`../waterfall/configuration` page.
```
```suggestion
To configure your proxy, see the :doc:`../waterfall/configuration` page.
```

Proximyst commented on source/waterfall/getting-started.rst:
```diff
@@ -0,0 +1,49 @@
+
+Click on the build number to download a file.
+
+Running The Server
```
```suggestion
Running The Proxy
```

Proximyst commented on source/waterfall/getting-started.rst:
```diff
@@ -0,0 +1,49 @@
+
+Waterfall and Travertine are drop in replacements for BungeeCord, you don't need to make any changes.
+
+Getting A Server Jar
```
```suggestion
Getting A Proxy Jar
```

Proximyst commented on source/waterfall/getting-started.rst:
```diff
@@ -0,0 +1,49 @@
+Getting A Server Jar
+~~~~~~~~~~~~~~~~~~~~~
+
+Paper provides runnable server jars directly from our `website's downloads page <https://papermc.io/downloads#Waterfall>`_.
```
```suggestion
Paper provides runnable proxy jars directly from our `website's downloads page <https://papermc.io/downloads#Waterfall>`_.
```

Proximyst commented on source/waterfall/getting-started.rst:
```diff
@@ -0,0 +1,49 @@
+.. contents::
+   :depth: 1
+   :local:
+
```
Section on what Waterfall and Travertine *are*. Note that Travertine is Waterfall but 1.7 support.

Chew replied:
Specifically 1.7.10 support ^

Proximyst commented on source/waterfall/getting-started.rst:
```diff
@@ -0,0 +1,49 @@
+Running The Server
+~~~~~~~~~~~~~~~~~~
+
+To run the server, simply start it up like any other Java application.
```
```suggestion
To run the proxy, simply start it up like any other Java application.
```

Proximyst commented on source/waterfall/getting-started.rst:
```diff
@@ -0,0 +1,49 @@
+
+To configure your server, see the :doc:`../waterfall/configuration` page.
+
+Updating The Server
```
```suggestion
Updating The Proxy
```

Proximyst commented on source/waterfall/getting-started.rst:
```diff
@@ -0,0 +1,49 @@
+Updating The Server
+~~~~~~~~~~~~~~~~~~~
+
+To update the server, first stop it safely by executing the ``end`` command
```
```suggestion
To update the proxy, first stop it safely by executing the ``end`` command
```

Proximyst commented on source/waterfall/getting-started.rst:
```diff
@@ -0,0 +1,49 @@
+~~~~~~~~~~~~~~~~~~~
+
+To update the server, first stop it safely by executing the ``end`` command
+and then replace the old waterfall jar with a new one. That's it.
```
```suggestion
and then replace the old proxy jar with a new one. That's it.
```

This is an article for two different proxies.

finlaysawyer commented:
I've made the minor changes in regards to grammar etc and added a section at the top explaining what Waterfall and Travertine are. I have added back the link to Aikar's post, but still leaving the Waterfall flags above as they can only be found in Discord currently. I've also provided a better explanation about disabling entity metadata rewriting from what I could gather from issues & Discord, feel free to make any corrections. Cheers again
